src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/latest.sql

```diff
@@ -106,6 +106,18 @@ BEGIN
     -- restore citus catalog tables
     --
     INSERT INTO pg_catalog.pg_dist_partition SELECT * FROM public.pg_dist_partition;
+
+    -- if we are upgrading to PG16, we need to add varnullingrels to partkey
+    IF substring(current_Setting('server_version'), '\d+')::int = 16 THEN
```

Comment:
what if the user upgrade from pg15 to pg17? I don't think pg_upgrade prevents that?

I think we could make this more generic. We iterate on all the vars in pg_dist_partition, and if misses `varnullingrels` then we do this.

Reply:
Yes, I also thought of that. Let me add `AND partkey NOT LIKE '%varnullingrels%';` and open a PR for this